### PR TITLE
Fix unit test of authentication.framework

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Fixed existing unit tests for authentication.framework

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandlerTest.java [hunk 1]
```diff
@@ -173,6 +171,7 @@ public void testGetServiceProviderMappedUserRoles(Map<String, String> spRoleMapp
 
     @DataProvider(name = "spRoleClaimUriProvider")
     private Object[][] getSpRoleClaimUriData() {
+        Util.mockIdentityUtil();
```
Comment (Thumimku) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandlerTest.java:174]: It's good we have one line between method signature and logic. Please check other places too.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/session/extender/request/SessionExtenderRequestFactoryTest.java [hunk 2]
```diff
@@ -106,7 +103,7 @@ public void testCanHandle(String requestURI, boolean canHandleResult) {
         return new Object[][]{
                 {IDP_SESSION_KEY, null, IDP_SESSION_KEY, null},
                 {null, new Cookie[] {sessionCookie}, null, sessionCookie},
-                {IDP_SESSION_KEY, new Cookie[] {sessionCookie}, IDP_SESSION_KEY, null}
+                {IDP_SESSION_KEY, new Cookie[] {sessionCookie}, IDP_SESSION_KEY, sessionCookie}
```
Comment (Thumimku) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/session/extender/request/SessionExtenderRequestFactoryTest.java:106]: Why we change null to sessionCookie?

Comment (Thumimku) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/session/extender/request/SessionExtenderRequestFactoryTest.java:106]: If needed we can have additional data entry. no need to manipulate existing one.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/Util.java [hunk 3]
```diff
@@ -96,4 +98,9 @@ public static ClaimHandler mockClaimHandler() throws FrameworkException {
                 any(Map.class), anyBoolean());
         return claimHandler;
     }
+
+    public static void mockIdentityUtil(){
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.getLocalGroupsClaimURI()).thenReturn(UserCoreConstants.ROLE_CLAIM);
```
Comment (Thumimku) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/Util.java:104]: can we specify why this is needed?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/session/extender/request/SessionExtenderRequestFactoryTest.java (hunk 2 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/session/extender/request/SessionExtenderRequestFactoryTest.java:106]: There is an issue with the test case (dataProvider). 
When the sessionExtenderRequestFactory.create method called with the parameter mockedHttpRequest where, 
        (mockedHttpRequest.getCookies()).thenReturn(requestCookies);
The builded resultant object gives the corresponding cookie (here sessionCookie, not null) when getSessionCookie() method calls.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/Util.java (hunk 3 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/Util.java:104]: To address the issue of java.lang.ExceptionInInitializerError related to the IdentityUtil.getLocalGroupsClaimURI()

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandlerTest.java (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandlerTest.java:178]: Addressed with commit 1ab06b94b3607a2e2f94d5cfb23dbf0d5074d3cb

**Review by Thumimku — APPROVED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandlerTest.java [hunk 4]
```diff
@@ -89,10 +88,9 @@
 import static org.testng.Assert.fail;
 import static org.wso2.carbon.identity.core.util.IdentityUtil.getLocalGroupsClaimURI;
 
-@PrepareForTest({
-                        FrameworkUtils.class, IdentityApplicationManagementUtil.class, ApplicationMgtSystemConfig.class,
-                        IdentityTenantUtil.class
-                })
+@PrepareForTest({FrameworkUtils.class, IdentityApplicationManagementUtil.class, ApplicationMgtSystemConfig.class,
+        IdentityTenantUtil.class, IdentityUtil.class})
+
```
Comment (sachiniWettasinghe) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandlerTest.java:93]: Remove redundant line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/session/extender/request/SessionExtenderRequestFactoryTest.java [hunk 5]
```diff
@@ -54,7 +53,8 @@
 /**
  * Unit test cases for SessionExtenderRequestFactory.
  */
-public class SessionExtenderRequestFactoryTest extends PowerMockTestCase {
+
```
Comment (sachiniWettasinghe) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/session/extender/request/SessionExtenderRequestFactoryTest.java:56]: Remove redundant line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandlerTest.java (hunk 4 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandlerTest.java:93]: Addressed with commit 12e680f42aedfa730f6f23a0c28f99c8a56868da in https://github.com/wso2/carbon-identity-framework/pull/3774 PR

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/session/extender/request/SessionExtenderRequestFactoryTest.java (hunk 5 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/session/extender/request/SessionExtenderRequestFactoryTest.java:56]: Addressed with commit 12e680f42aedfa730f6f23a0c28f99c8a56868da in https://github.com/wso2/carbon-identity-framework/pull/3774 PR

**Pull request merged**